### DISCUSSION TITLE
Heads up about 4.0 when it comes to serialization support

### DISCUSSION THREAD

**Discussion by brizzbuzz** [bkbnio/kompendium#503] [General]
Hi all.  Just a general warning.  I'm not sure when Kompendium 4 will come out, but, when it does, it is very likely that the only serializer that will be supported out of the box will be Kotlinx Serialization. 

Currently, Kompendium relies on `SchemaConfigurator` implementations to offer customizable support for each serialization framework (Jackson, Gson, etc. etc.). I quite like this approach, and will most likely keep it, in order to provide users of other serialization libraries a way to provide the needed functionality. 

However, it is too much of a burden to try to implement features for all of them, so every implementation except for Kotlinx serialization will most likely be removed.  Kotlinx serialization will be considered the "golden path" for Kompendium, and if you want to use another framework, it will be on you / the community to maintain configurator implementations. 

Hope this does not cause too much strife or burdern 🙏 

TL;DR Kompendium will keep support for custom serialization configurators, but, KotlinX will become the default configurator, and all others will be removed from the library.
